crates/librqbit_core/src/magnet.rs

```diff
@@ -45,20 +65,52 @@ impl Magnet {
 
 impl std::fmt::Display for Magnet {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(
-            f,
-            "magnet:?xt=urn:btih:{}&tr={}",
-            self.info_hash.as_string(),
-            self.trackers.join("&tr=")
-        )
+        if let (Some(id20), Some(id32)) = (self.id20, self.id32) {
```

Comment:
nit: it's a bit repetitive to handle all cases like this, you could have done this without a big branch, but small branches instead:

```
write!(f, "magnet:?")?;
if let Some(id20) = self.id20 {
    write!("&...");
}
if let Some(id32) = self.i32 {
    write!("&...");
}
write!("&tr={}", self.trackers.join("&tr="))?;
```